torchtitan/models/attention.py

```diff
@@ -25,7 +25,9 @@
 # FlexAttention mask type. For each mask type, we initialize it at most once per
 # batch. To record what it is initialized, FLEX_ATTN_MASK_T is used as the key to
 # track the initialized mask.
-FLEX_ATTN_MASK_T = tuple[str, int | None]
+FLEX_ATTN_MASK_T = tuple[
+    str, int | None, int | None
+]  # (mask_type, fixed_block_size, sliding_window)
 
 
 class FlexAttention(torch.nn.Module):
```

Comment:
@wwwjn  will rebase onto #1776

Reply:
Sorry for the disruption. I should have done this earlier.

As for the FlexAttention, @drisspg confirmed that, while it is probably just a minor overhead, the AuxOutput does incur some extra memory and memory write. So let's keep it optional.

Reply:
Sure, I will keep AuxOutput as optional